src/main/scala/guessbook/GuessBookUserInput.scala

```diff
@@ -0,0 +1,206 @@
+            }
+            case AuthorGender => {
+              val guessValue: GenderEnum.Value = specifyEnumValue(GenderEnum)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case CountryOfOrigin => {
+              val guessValue: Country.Value = specifyEnumValue(Country)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case DetectiveType => {
+              val guessValue: Detective.Value = specifyEnumValue(Detective)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+            case SuspectPoolType => {
+              val guessValue: SuspectPool.Value = specifyEnumValue(SuspectPool)
+              booksOnBoard(guesser-1) = game.poseQuestion(selectedBooks(guesser-1), guessAttribute, guessValue, booksOnBoard(guesser-1))
+            }
+          }
+
+          println(s"Books remaining on P$guesser's board:")
+          game.printRemainingBooks(booksOnBoard(guesser-1), maxBooksPerLine = 5)
+        }
+        case 2 => { // guessing book title
+          val guessedTitle: String = specifyBookTitle()
+          val guessResult = game.guessBookTitle(selectedBooks(guesser-1), guessedTitle, guesser, booksOnBoard(guesser-1))
+          booksOnBoard(guesser-1) = guessResult._1
+          gameOver = guessResult._2
+          if (!gameOver){
+            println(s"Books remaining on P$guesser's board:")
+            game.printRemainingBooks(booksOnBoard(guesser-1), maxBooksPerLine = 5)
+          }
+        }
+      }
+    }
+
+    println("END OF GAME")
+  }
+
+  def selectBookForOpponent(guesser: Int, numBlankLines: Int = 25): Book = {
+    val game = GuessBook
+    val nonGuesser = guesser % 2 + 1
+    var matchingBook: Option[Book] = None
+
+    println(s"Player $guesser, select a book for Player $nonGuesser to guess.")
+    println(s"Player $nonGuesser, please look away.\n")
+    println("The available books are:")
+    game.printRemainingBooks(game.allBooks, maxBooksPerLine = 5)
+    println()
+
+    while (matchingBook.isEmpty) {
+      println(s"Enter the title of the book for Player $nonGuesser to guess:")
+      val titleInput = StdIn.readLine()
+      matchingBook = game.allBooks.find(book => book.title.toLowerCase == titleInput.toLowerCase)
+      if (matchingBook.isEmpty)
+        println(s"$titleInput is not the title of a book in the game. Please try again.\n")
+    }
+
+    println("Book selected." + "\n" * numBlankLines)
+    matchingBook.getOrElse(throw new Exception("Book should have been selected already"))
+  }
+
+  def chooseAttributeOrTitle(): Int = {
+    var input: String = ""
+    var validInput = false
```

Comment:
VARS?!?!?

Reply:
Changed all such methods taking user input to use recursion!